packages/babel-plugin-transform-react-constant-elements/test/fixtures/constant-elements/append-to-end-when-declared-in-scope-2/output.mjs

```diff
@@ -1,17 +1,25 @@
 var _ref =
 /*#__PURE__*/
-<div>child</div>;
+(() => <div>child</div>)();
 
 const AppItem = () => {
   return _ref;
 };
 
+var _ref3 =
+/*#__PURE__*/
+(() => <p>Parent</p>)();
+
+var _ref4 =
+/*#__PURE__*/
+(() => <AppItem />)();
+
 var _ref2 =
 /*#__PURE__*/
-<div>
-        <p>Parent</p>
-        <AppItem />
-      </div>;
+(() => <div>
+        {_ref3}
+        {_ref4}
+      </div>)();
```

Comment:
This should be kept as a single element, not as 3 separate elements.
The arrow function is not needed if the elements don't have impure children anyway.